java/arcs/android/storage/database/DatabaseImpl.kt

```diff
@@ -110,7 +111,8 @@ class DatabaseImpl(
     ) = when (dataType) {
         DatabaseData.Entity::class -> getEntity(storageKey, schema)
         DatabaseData.Collection::class -> getCollection(storageKey, schema)
-        else -> TODO("Support Singletons")
+        DatabaseData.Singleton::class -> getSingleton(storageKey, schema)
+        else -> throw UnsupportedOperationException("Unsupported data type $dataType.")
```

Comment:
DatabaseData is a sealed class, you probably don't need this `else`.

Reply:
Done

Reply:
Actually, not done. It's not smart enough to know that it is exhaustive (compile error). I think because it's a KClass rather than an actual DatabaseData instance.